src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -154,7 +174,7 @@ private static void analyzeRepos(String outputPath, List<RepoConfiguration> conf
             } catch (RuntimeException e) {
                 logger.log(Level.SEVERE, String.format(MESSAGE_BRANCH_DOES_NOT_EXIST,
                         config.getBranch(), config.getLocation()), e);
-                generateEmptyRepoReport(repoReportDirectory.toString());
+                handleBranchingFailed(config);
```

Comment:
We may need to create more specific exceptions.

So you can just catch and add to `ErrorSummary`. This way you can handle them generally.

Also, we can then store `LOG_ERROR_CLONING`, `LOG_BRANCH_DOES_NOT_EXIST` and each specific error messages in its exception class.

Start by creating two exceptions? 

In the future, we can group them as print only, clean up, etc.

Reply:
if this is difficult then let's do it in another PR.

Its because `ProcessRunner` throws general `RuntimeException`. If you were to create some specific git exception, we need tests too.

Reply:
I think it may be hard to do it at this point, as PR #765 is revamping the entire cloning process. 
Should we wait for #765 to merge in before working on individual exception classes in a future PR?

Reply:
Yup, the PR is quite large already.

Please create an issue on this.